code/modules/events/brand_intelligence.dm

```diff
@@ -29,6 +39,13 @@
 
 /datum/event/brand_intelligence/tick()
 	if(!vendingMachines.len || !originMachine || originMachine.shut_up)	//if every machine is infected, or if the original vending machine is missing or has it's voice switch flipped
+		//VORESTATION Add - Effects when 'source' machine is destroyed/silenced
+		for(var/obj/machinery/vending/saved in infectedVendingMachines)

```

Comment:
1) That isn't a typeless loop
2) On line 49, it calls end()     end() does exactly what you do here, so you have looped over infectedVendingMachines twice in a row.